src/rdb_protocol/terms/writes.cc

```diff
@@ -278,7 +278,7 @@ class replace_term_t : public op_term_t {
             rcheck(args->arg_is_deterministic(1) == deterministic_t::always,
                    base_exc_t::LOGIC,
                    "Could not prove argument deterministic.  "
-                   "Maybe you want to use the non_atomic flag?");
+                   "Maybe you want to use the non_atomic flag in the Update function");
```

Comment:
Any suggestions @AmeyDeshpande ? Do you want to remove this change from this PR for now, so we can merge the other one?